docs/theming.md

```diff
@@ -183,6 +183,161 @@ pod 'MaterialComponents/schemes/Color'
 ```
 <!--</div>-->
 
+## Migration guides
+
+### Migration guide: Themers to Theming extensions
+
+Themers will eventually be deprecated and deleted. Theming extensions are the recommended
+replacement. This migration guide will cover the typical migration path from Themer usage to
+Theming extensions.
+
+In general, every component's `Themer` targets will gradually be replaced by a single `Theming`
+extension target. This includes:
+
+- ColorThemer
+- FontThemer
+- ShapeThemer
+- TypographyThemer
+- Component Themers (e.g. CardThemer)
+
+#### Typical migration
+
+When migrating from a subsystem Themer to a Theming extension the first thing to understand is that
+a Theming extension will apply *all* of the available subsystems to a given component. If you were
+previously relying on the ability to apply only one subsystem (e.g. Color) to a component, please
+file a
+[feature request](https://github.com/material-components/material-components-ios/issues/new/choose)
+with a code snippet of your existing use case.
+
+Generally-speaking, the migration from a subsystem Themer to a Theming extension will involve the
+following steps:
+
+1. Update your dependencies.
+2. Update your imports.
+3. Changes to your code.
+
+##### Update your dependencies
+
+In order to use theming extensions you'll need to first follow the
+[beta component instructions](../../contributing/beta_components.md).
```

Comment:
For migration, should we document that there are some theming extension available in non-beta, and some are in beta? My impression from it would be people have to add beta dependency to use it.

Reply:
Good call; done.